include/mrdox/Metadata/Typedef.hpp

```diff
@@ -30,15 +33,29 @@ struct TypedefInfo : SymbolInfo
     //   typedef std::vector<int> MyVector;
     bool IsUsing = false;
 
+    std::unique_ptr<TemplateInfo> Template;
+
     //--------------------------------------------
 
     static constexpr InfoType type_id = InfoType::IT_typedef;
 
+    explicit
     TypedefInfo(
         SymbolID id_ = SymbolID::zero)
         : SymbolInfo(InfoType::IT_typedef, id_)
     {
     }
+
+private:
+    // KRYSTIAN NOTE: if Template is non-null,
+    // then IsUsing *must* be set; should we do it here?
```

Comment:
Maybe `IsUsing` should be a private data member and we would provide a public member function `isAlias()`